### PR TITLE
Add login, register and address autocomplete in checkout

### PR DESCRIPTION
**Pull request by JaroslavSku**

### CODE REVIEW

File: modules/checkout-magento/services/useEmailAvailable.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import useGetIsEmailAvailable from '#ioc/repositories/useGetIsEmailAvailable'
```
Comment (adamjedlicka) [inline modules/checkout-magento/services/useEmailAvailable.ts:1]: Repositories have Repository suffix

File: modules/theme-tailwind/organisms/CheckoutLoginModal.vue [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  email: {
+    type: String,
+    required: true,
+  },
+})
+
+const onSubmit = async ({ password }: { password: string }) => {
+  try {
+    isLoading.value = true
+    await loginCustomer(props.email, password, { redirect: false })
```
Comment (adamjedlicka) [inline modules/theme-tailwind/organisms/CheckoutLoginModal.vue:43]: This is framework logic that should be encapsulated in some service (loginCustomerInCheckout). Why should programmer that wants to login customer in checkout know they shoudl disable the redirect and patch the store?

File: modules/theme-tailwind/organisms/CheckoutRegisterModal.vue [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  email: {
+    type: String,
+    required: true,
+  },
+})
+
+const onSubmit = async (data: any) => {
+  try {
+    loading.value = true
+    await registerCustomer(data, {
```
Comment (adamjedlicka) [inline modules/theme-tailwind/organisms/CheckoutRegisterModal.vue:71]: Same problem here. Extract to registerCustomerInCheckout service

File: modules/theme-tailwind/organisms/ContactInfoSelection.vue [hunk 4]
```diff
@@ -39,44 +59,134 @@
...
+      <div class="mt-4 flex items-end">
+        <Button v-if="offerLogin" color="primary" @click="showLogin = true">{{ t('Login') }}</Button>
+        <Button v-if="offerRegistration" color="primary" @click="showRegistration = true">{{ t('Register') }}</Button>
+      </div>
     </Form>
+
+    <CheckoutLoginModal
+      v-if="showLogin"
+      :email="contactEmail"
+      @close="
```
Comment (adamjedlicka) [inline modules/theme-tailwind/organisms/ContactInfoSelection.vue:71]: Extract to named function

File: modules/theme-tailwind/organisms/ContactInfoSelection.vue [hunk 5]
```diff
@@ -39,44 +59,134 @@
...
+        () => {
+          showLogin = false
+          offerLogin = false
+        }
+      "
+    />
+    <CheckoutRegisterModal
+      v-if="showRegistration"
+      :email="contactEmail"
+      @close="
```
Comment (adamjedlicka) [inline modules/theme-tailwind/organisms/ContactInfoSelection.vue:81]: Extract to named function

File: modules/theme-tailwind/organisms/ContactInfoSelection.vue [hunk 6]
```diff
@@ -39,44 +59,134 @@
...
     countryCode: 'CZ',
-    ...data,
+    ...address,
   })
 
   emit('confirm')
 }, 500)
 
+onMounted(async () => {
+  const { addresses } = await getCustomerAddresses()
```
Comment (adamjedlicka) [inline modules/theme-tailwind/organisms/ContactInfoSelection.vue:180]: Why is this done as separate request in mounted instead done as part of the initial checkout request?

File: modules/theme-tailwind/organisms/ContactInfoSelection.vue [hunk 7]
```diff
@@ -39,44 +59,134 @@
...
+}
 
 const form = ref<any>(null)
 
 const onInput = debounce(async (data: any) => {
-  if (!form.value?.isValid) return emit('select')
+  if (!form.value?.isValid && !selectedAddress.value) return emit('select')
+  const address = selectedAddress.value ? selectedAddress.value : data
+
+  contactEmail.value = form.value?.getData()?.email ?? {}
```
Comment (adamjedlicka) [inline modules/theme-tailwind/organisms/ContactInfoSelection.vue:164]: You already have the form data as part of this function input arguments. No need to get it from form again.

**Review by adamjedlicka — CHANGES_REQUESTED** [review]

File: modules/theme-tailwind/organisms/ContactInfoSelection.vue (hunk 6 above)
Comment (JaroslavSku) [reply modules/theme-tailwind/organisms/ContactInfoSelection.vue:180]: We can discuss it when you are online, I use the addresses in watch and when I use useAsyncData then it tells me the addresses are undefined.

**Review by adamjedlicka — APPROVED** [review]

**Pull request merged**